tools/PHP_CLI_Helper.class.inc

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Use bash to determine if certain unix tools are installed.  Which is used for
+ * generic system tools (e.g. wget), and dpkg if which fails.
+ *
+ * @param string $tool Name of tool
+ *
+ * @return bool representing if tool is installed
+ */
+function installed($tool) : bool
+{
+    // `which` returns empty if a tool is not installed.
+    // shell_exec captures this output.
+    if (shell_exec("which $tool")) {
+        return true;
+    }
+    // check installed pacakages with dpkg. Returns 0 if installed.
+    exec("dpkg -s $tool", $output, $status);
+    if ($status === 0) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Prompts a user with a question and acceptable responses.
+ *
+ * @param string $question Prompt to display to user
+ * @param array  $answers  List of acceptable answers to prompt
+ *
+ * @return void
+ */
+function writeQuestion($question, $answers) : void
+{
+    echo $question . ' (' . implode('/', $answers) . '): ' . PHP_EOL;
+}
+
+/**
+ * Gets user input from STDIN and checks if it matches a valye in
+ * $possibleAnswers.  If not, the default answer is used.  Inteded to follow
+ * function writeQuestion
+ *
+ * @param array $possibleAnswers Possible answers to a prompt
+ * @param array $defaultAnswer   Response if user entered invalid response
```

Comment:
In most cases if a user responds with an invalid response, shouldn't the scripts be re-prompting for a valid answer?